plugins/pixart-rf/fu-pxi-wireless-device.c

```diff
@@ -0,0 +1,606 @@
+	self->fwstate.checksum += checksum;
+
+	for (guint i = 0; i < chunks->len; i++) {
+		FuChunk *chk2 = g_ptr_array_index (chunks, i);
+		if (!fu_pxi_wireless_device_write_payload (device, chk2, error))
+			return FALSE;
+		prn++;
+		/* check crc at fw when PRN over threshold write or
+		 * offset reach max object sz or write offset reach fw length */
+		if (prn >= self->fwstate.prn_threshold || i == chunks->len - 1) {
+			if (!fu_pxi_wireless_device_check_crc (device, self->fwstate.checksum, error))
+				return FALSE;
+			prn = 0;
+		}
+	}
+	/* success */
+	return TRUE;
+}
+
+static gboolean
+fu_pxi_wireless_device_fw_ota_init_new (FuDevice *device, gsize bufsz, GError **error)
+{
+	FuPxiWirelessDevice *self = FU_PXI_WIRELESS_DEVICE (device);
+	FuPxiReceiverDevice *parent = fu_pxi_wireless_device_get_parent (device ,error);
+	guint8 fw_version[10] = { 0x0 };
+	g_autoptr(GByteArray) ota_cmd = g_byte_array_new ();
+	g_autoptr(GByteArray) receiver_cmd = g_byte_array_new ();
+
+	fu_byte_array_append_uint8 (ota_cmd, 0X06);					/* ota init new command length */
+	fu_byte_array_append_uint8 (ota_cmd, FU_PXI_DEVICE_CMD_FW_OTA_INIT_NEW);	/* ota ota init new op code */
+	fu_byte_array_append_uint32 (ota_cmd, bufsz, G_LITTLE_ENDIAN);			/* fw size */
+	fu_byte_array_append_uint8 (ota_cmd, 0x0);					/* ota setting */
+	g_byte_array_append (ota_cmd, fw_version, sizeof(fw_version));			/* ota version */
+
+	/* increase the serial number */
+	self->sn++;
+
+	if (!fu_pxi_composite_receiver_cmd (FU_PXI_DEVICE_CMD_FW_OTA_INIT_NEW,
+							self->sn,
+							self->model.target,
+							receiver_cmd,
+							ota_cmd, error))
+		return FALSE;
+	if (!fu_pxi_wireless_device_set_feature (FU_DEVICE (parent), receiver_cmd->data, receiver_cmd->len, error))
+		return FALSE;
+
+	return TRUE;
+}
+
+
+static gboolean
+fu_pxi_wireless_device_fw_ota_ini_new_check (FuDevice *device, GError **error)
+{
+	FuPxiWirelessDevice *self = FU_PXI_WIRELESS_DEVICE (device);
+	FuPxiReceiverDevice *parent = fu_pxi_wireless_device_get_parent (device ,error);
+	guint8 buf[FU_PXI_RECEIVER_DEVICE_OTA_BUF_SZ] = { 0x0 };
+	guint8 status = 0x0;
+	g_autoptr(GByteArray) ota_cmd = g_byte_array_new ();
+	g_autoptr(GByteArray) receiver_cmd = g_byte_array_new ();
+
+	/* ota command */
+	fu_byte_array_append_uint8 (ota_cmd, 0x1);					/* ota command length */
+	fu_byte_array_append_uint8 (ota_cmd, FU_PXI_DEVICE_CMD_FW_OTA_INIT_NEW_CHECK);	/* ota command */
+
+	/* increase the serial number */
+	self->sn++;
+
+	/* get pixart wireless module ota command */
+	if (!fu_pxi_composite_receiver_cmd (FU_PXI_DEVICE_CMD_FW_OTA_INIT_NEW_CHECK,
+							self->sn,
+							self->model.target,
+							receiver_cmd,
+							ota_cmd, error))
+		return FALSE;
+	if (!fu_pxi_wireless_device_set_feature (FU_DEVICE (parent), receiver_cmd->data,
+							receiver_cmd->len, error))
+		return FALSE;
+
+	/* delay for wireless module device get command response*/
+	g_usleep (50 * 1000);
```

Comment:
this same delay is used in a lot of places, I think it should be a define

Reply:
Ok it will be defined

Reply:
has added define.